src/frontc/cparser.mly

```diff
@@ -954,6 +954,7 @@ decl_spec_list:                         /* ISO 6.7 */
                                         /* ISO 6.7.4 */
 |   INLINE decl_spec_list_opt           { SpecInline :: $2, $1 }
 |   NORETURN decl_spec_list_opt         { SpecNoreturn  :: $2, $1 }
+|   ATOMIC LPAREN decl_spec_list_opt RPAREN { SpecCV(CV_ATOMIC) :: $3, $1 }
```

Comment:
This doesn't seem right. All the other cases end with `decl_spec_list_opt`, which allow other specifiers to appear after them, but this one pretends that whatever is in the parenthesis is actually after it (consed to the atomic specifier) and after `RPAREN` no specifiers are allowed.

Moreover, 6.7.2.4 in C11 standard says the argument in the parenthesis is "type-name", not just other specifiers.
I think there's some fundamental difference between the type specifier and type qualifier versions of `_Atomic` that this isn't correctly capturing. Otherwise there wouldn't be any need to have both in the standard.

Reply:
According to https://en.cppreference.com/w/c/language/atomic there does not really seem to be a difference.

> 1) Use as a type specifier; this designates a new atomic type
> 2) Use as a type qualifier; this designates the atomic version of type-name. In this role, it may be mixed with const, volatile, and restrict), although unlike other qualifiers, the atomic version of type-name may have a different size, alignment, and object representation.

But I guess you are right about allowing other attributes to come after.

Reply:
Just directly reading the corresponding grammar from the standard there is a difference: the type specifier version takes a "type-name" argument, which in addition to "specifier-qualifier-list" (which this currently does handle) may also optionally have an "abstract-declarator" with some kind of pointer stuff.

Reply:
Yes, it seems like it. I now relaxed it a bit further, but `_Atomic(int*)` for example still does not parse...
Not sure if we want to fix that or also add an issue for it and do it later, as it will probably require changes to CIL as well to properly represent this.

Reply:
Probably can be left into a separate issue if there's no immediate need for it to parse any project.